### PR TITLE
Ensure that non-Dart-2 pages are marked as such

### PR DESCRIPTION
**Pull request by chalin**
This PR also contains some minor clearnup.

Contributes to https://github.com/dart-lang/site-www/issues/870#issuecomment-401070995.

Staged, see, e.g.: 
- https://dartlang-org-staging-0.firebaseapp.com/guides/language/sound-dart#simple-type-assignment
- https://dartlang-org-staging-0.firebaseapp.com/articles/dart-vm/native-extensions et all are now marked as "might be obsolete".

cc @munificent

### CODE REVIEW

File: src/_articles/dart-vm/native-extensions.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ title: "Native Extensions for the Standalone Dart VM"
 description: "Learn how to enable command-line Dart apps to call C/C++ functions."
 written: 2012-05-01
 category: dart-vm
+obsolete: true
```
[the remark this answers is not in the document]
Comment (a-siva) [reply src/_articles/dart-vm/native-extensions.md]: Native extensions continues to work in Dart2, as long as the page doesn't say this feature is obsolete I am ok.

[the remark this answers is not in the document]
Comment (chalin) [reply src/_articles/dart-vm/native-extensions.md]: I've dropped "obsolete: true", and will update the page in a followup PR.

**Comment by chalin** [comment]
Edits done and staging server refreshed. PTAL @kwalrath

**Comment by kwalrath** [comment]
I'd still like to have more helpful obsolescence text, but it's not worth holding up this PR.

**Pull request merged**